powerdns/tests/test_api.py

```diff
@@ -62,21 +68,195 @@ def test_record_filters_by_ip_and_type(self):
+        domain = DomainFactory(name='example.com', owner=self.regular_user1)
+        data = {
+            'type': 'cname'.upper(),
+            'domain': '/api/domains/' + str(domain.id) + '/',
+            'name': 'example.com',
+            'content': '192.168.0.1',
+        }
+        response = self.client.post(
+            reverse('record-list'), data, format='json',
+            **{'HTTP_ACCEPT': 'application/json; version=v2'}
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
-        url = reverse('domain-detail', args=(domain_id,))
-        response = self.client.get(url, format='json')
+    def test_record_is_not_created_when_domain_is_other(self):
+        self.client.login(username='regular_user1', password='regular_user1')
+        domain = DomainFactory(name='example.com', owner=self.regular_user2)
+        data = {
+            'type': 'cname'.upper(),
+            'domain': '/api/domains/' + str(domain.id) + '/',
+            'name': 'example.com',
+            'content': '192.168.0.1',
+        }
+        response = self.client.post(
+            reverse('record-list'), data, format='json',
+            **{'HTTP_ACCEPT': 'application/json; version=v2'}
+        )
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+
+    #
+    # updates
+    #
+    def test_update_record_when_recordrequest_doesnt_exist(self):
+        self.client.login(username='super_user', password='super_user')
+        record = RecordFactory(
+            auto_ptr=AutoPtrOptions.NEVER.id,
+            type='A',
+            name='blog.com',
+            content='192.168.1.0',
+        )
+        new_name = 'new-' + record.name
+        response = self.client.patch(
+            reverse('record-detail', kwargs={'pk': record.pk}),
+            data={'name': new_name},
+            format='json',
+            **{'HTTP_ACCEPT': 'application/json; version=v2'}
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        record.refresh_from_db()
+        self.assertEqual(record.name, new_name)
+        self.assertEqual(
+            RecordRequest.objects.filter(record__id=record.id).count(), 1
+        )
+
+    def test_update_record_when_recordrequest_exists(self):
+        self.client.login(username='super_user', password='super_user')
+        record_request = RecordRequestFactory(
+            record__auto_ptr=AutoPtrOptions.NEVER.id,
+            record__type='A',
+            record__name='blog.com',
+            record__content='192.168.1.0',
+        )
+        new_name = 'new-' + record_request.record.name
+        response = self.client.patch(
+            reverse('record-detail', kwargs={'pk': record_request.record.pk}),
+            data={'name': new_name},
+            format='json',
+            **{'HTTP_ACCEPT': 'application/json; version=v2'}
+        )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
when there is already created request for some record, shouldn't we reject next request(s), until the first one will be finished?